endpoints-common/src/main/scala/com/thenewmotion/ocpi/common/AuthorizedRequests.scala

```diff
@@ -18,11 +17,26 @@ trait AuthorizedRequests {
   private val logRequest: HttpRequest => HttpRequest = { r => logger.debug(r.toString); r }
   private val logResponse: HttpResponse => HttpResponse = { r => logger.debug(r.toString); r }
 
-  protected def requestWithAuth(http: HttpExt, req: HttpRequest, auth: AuthToken[Ours])
-    (implicit ec: ExecutionContext, mat: Materializer): Future[HttpResponse] = {
-    http.singleRequest(logRequest(req.addCredentials(GenericHttpCredentials("Token", auth.value, Map())))).map { response =>
+  private def requestWithAuthSupportingRedirect(
+    http: HttpExt, req: HttpRequest, auth: AuthToken[Ours], redirectCount: Int = 0
+  )(implicit ec: ExecutionContext, mat: Materializer): Future[HttpResponse] = {
+    http.singleRequest(
+      logRequest(req.addCredentials(GenericHttpCredentials("Token", auth.value, Map())))
+    ).flatMap { response =>
       logResponse(response)
-      response
+      response.status match {
+        case StatusCodes.PermanentRedirect | StatusCodes.TemporaryRedirect if redirectCount < 10 =>
+          response.header[Location].map { newLoc =>
+            logger.warn("Following redirect to {}", newLoc.uri)
+            response.discardEntityBytes()
```

Comment:
what would happen without this?

Reply:
See https://doc.akka.io/docs/akka-http/current/implications-of-streaming-http-entity.html

>>> Warning
Consuming (or discarding) the Entity of a request is mandatory! If accidentally left neither consumed or discarded Akka HTTP will assume the incoming data should remain back-pressured, and will stall the incoming data via TCP back-pressure mechanisms. A client should consume the Entity regardless of the status of the HttpResponse.